563.Binary Tree Tilt

```diff
@@ -0,0 +1,33 @@
+/**
+ * Definition for a binary tree node.
+ * struct TreeNode {
+ *     int val;
+ *     TreeNode *left;
+ *     TreeNode *right;
+ *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
+ * };
+ */
+ //Time : O(n) Space : O(n)
```

Comment:
why space is O(n), please explain in detail.

Reply:
Every time we call the recursive function, we need some memory space. In the worst case, the height of the tree can be O(n), and we need memory for every node on that branch from root to leaf. So the space we need is O(n).

Reply:
when you say "some memory space", can you tell me what's in that memory?

Reply:
When we call a function we allocate some memory in the stack and some local variables are stored over there. In this case, the variable like "left_sum" need space and are stored there.

Reply:
remember return value can also be store in stack.